### DISCUSSION TITLE
Config BGM volume while game is paused

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1234] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

I'm actually fond of the way the music pauses now when the game is paused, but it took some getting used to. I think people would appreciate being able to configure it to their liking.

### Describe the solution you'd like

A new setting in Audio Settings. No need for a full 0-100% slider, I think. Something like:

BGM during Pause (or something):
- Off (current behavior)
- Low (50% volume or so)
- Normal (0.98.2 behavior)

**Comment by CableDorado2** [comment]
I was going to suggest the same, so I support your comment. Also, it would be nice to have an option to Enable or Disable the BGM when the window is not in focus (when you minimize). Like KOF2002UM:

![20230803134648_1](https://github.com/ikemen-engine/Ikemen-GO/assets/18058378/d21be02e-bc5f-41d7-98a9-a850b4cbc885)

![20230803134651_1](https://github.com/ikemen-engine/Ikemen-GO/assets/18058378/8c346658-cde8-454b-9c10-4c1d636bd86b)

**Comment by Jesuszilla** [comment]
I think I may start work on this request sometime tomorrow since it's a holiday here in the U.S. I've already found what determines the pause BGM behavior so it shouldn't be hard at all.

While I'm at it, should I also refactor this so that it does the same operations on all sound channels as well? I find it rather annoying that looping sounds continue through pause when I'm pretty sure we can do something about it now.

**Reply by potsmugen** [reply]
That sounds great actually. I just had a look at a couple Capcom games. In CvS2 a pause stops all the sounds. In SF4 and SF5 it just pauses them. The latter is definitely cooler.

**Comment by shinra358** [comment]
Also if you want to use a different music altogether or a random range of user defined tracks every time you enter (that would be located in a standardized pause range folder included in the char files.

**Reply by Jesuszilla** [reply, marked as answer]
Assuming you have the `PauseMasterVolume` set to a value greater than 0, you should be able to modify the Lua to add a `playBGM` combined with `getStageInfo` (to restore the original track later) to get this behavior currently.

If `getStageInfo` is not sufficient then I propose we modify the `playBGM` Lua method to return the BGM object so that any BGM played through Lua (which should control all BGM outside of command line fights) can have its reference stored so that its details can be recalled in Lua later.

**Comment by K4thos** [comment]
PauseMasterVolume was already implemented by Jesuszilla some time ago.
